distributed/client.py

```diff
@@ -1053,6 +1057,26 @@ def _update_scheduler_info(self):
         except EnvironmentError:
             logger.debug("Not able to query scheduler for identity")
 
+    @gen.coroutine
+    def _wait_until_n(self, func, n):
+        while n and (yield func()) < n:
+            yield gen.sleep(0.2)
+
+    def wait_until_n(self, workers=0):
+        """
+        Note: If you add more arguments, you will need to return a list of
+        futures, or assert that only 1 is called at a time.
+        """
+        if workers:
+            @gen.coroutine
+            def f():
+                info, _ = self._get_current_info_and_scheduler()
```

Comment:
Thoughts on using `yield self.scheduler.identity()` here instead ?

Reply:
`_get_current_info_and_scheduler` helps make sure that we have a scheduler and a cluster, and that they are valid. it's a wrapper around the first part of `_repr_html_`.

your suggested way seems fine to me, and was the first way I did it, but I was just doing what @guillaumeeb suggested.  I definitely am less aware of all of the possible scenarios than you guys.

> Maybe you will need to add some conditionals like in 

is there no need to worry about that?

Reply:
Please just use `yield self.scheduler.identity()`.  It's simpler and cleaner.  If you run into an issue with this then bring it up and we'll handle it.  I think that you'll be fine though.

> helps make sure that we have a scheduler and a cluster, and that they are valid

We may not ever have a cluster object locally.   That shouldn't be required.  

If you want to wait until things are set up cleanly then you could `yield self`, but that should already have run, and I don't think should be necessary.